### DISCUSSION TITLE
$wire or @entangle does not work when installing alpinejs via npm

### DISCUSSION THREAD

**Discussion by AntonioDiPassio-AppSys** [alpinejs/alpine#1376] [7. Bugs (archived)]
If I use alpine via the CDN, $wire and @entangle work as described, but not when alpine is installed through npm.

Steps to reproduce:
- Install alpine: `npm i alpinejs`
- Add line to end of bootstrap.js: `import('alpinejs');`
- In <head>: `<script src="{{ mix('js/app.js') }}" defer></script>`
- Create new livewire component:
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class Test extends Component
{
    public $var = 'OK';

    public function render()
    {
        return view('livewire.test');
    }
}
```
- Blade view of component:
```
 <div x-data>
    <p>var: <span x-text="$wire.var"></span></p>
</div>
```

**Comment by joshhanley** [comment]
@AntonioDiPassio-AppSys check which version of Alpine is being installed by npm, run `npm list --depth=0` and see what Alpine says. If it's installing v3, then it's not currently compatible with Livewire, so you will need to install v2.

Hope this helps!

**Reply by kurucu** [reply]
Any view on when this will be?

**Reply by joshhanley** [reply]
Probably a couple of weeks. See here https://github.com/alpinejs/alpine/discussions/1426#discussioncomment-860213

**Reply by kurucu** [reply]
I didn’t see that. Thanks!

**Reply by joshhanley** [reply]
@kurucu support for Alpine V3 has been added in Livewire 2.5.0 today 🙂

**Comment by AntonioDiPassio-AppSys** [comment]
@joshhanley Version of alpinejs installed: `+-- alpinejs@2.8.2`
Doesn't matter now, I'll wait for v3 to support livewire

**Reply by joshhanley** [reply]
@AntonioDiPassio-AppSys ah interesting. Can you share the error you are getting?

Also support for Alpine V3 has been added in Livewire 2.5.0 today 🙂

**Comment by AntonioDiPassio-AppSys** [comment]
@joshhanley No error is displayed, it just fails silently

**Reply by kurucu** [reply]
Same here :(

**Reply by joshhanley** [reply]
@AntonioDiPassio-AppSys or @kurucu can each of you make sure Livewire and Alpine are both running properly?

In your browser console, check by typing `Alpine` or `Livewire` and hitting enter. Each should return an object. Alpine you can check the exact version in the browser console by typing `Alpine.version`.

If either Livewire or Alpine isn't running, please let me know. Also can you let me know what version of Alpine the browser is seeing?

**Reply by kurucu** [reply]
Both present. Alpine is v3.0.6 (and composer says v2.5.0 for Livewire, just to be sure!)

**Reply by joshhanley** [reply]
@kurucu ok great, thanks for confirming! The import process has changed on Alpine V3, you need to change to the following

```js
import Alpine from 'alpinejs'

window.Alpine = Alpine

Alpine.start()
```

See here in the docs https://alpinejs.dev/essentials/installation#as-a-module

Hope this helps!

**Reply by kurucu** [reply]
I had already done that. Then remembered I'd commented out the `@entangle(...)` to avoid errors. Then found I had the property marked as protected. So to cut a long story short it actually works and it's all my fault. Thank you VERY much for your help everyone. I hope @AntonioDiPassio-AppSys finds resolution soon, too.

**Reply by AntonioDiPassio-AppSys** [reply]
Livewire and Alpine are loaded in the browser. Alpine version is 2.8.2
They are working properly except for the $wire bindings.

I just tried this code in my component:

Test.php
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class Test extends Component
{
    public $var = 'init';

    public function test(){
        $this->var = 'click';
    }

    public function render()
    {
        return view('livewire.test');
    }
}
```

test.blade.php
```
<div x-data>
    <p>x-text: <span x-text="$wire.var"></span></p>
    <button @click="$wire.test()">$wire</button>

    <p>blade: <span>{{ $var }}</span></p>
    <button wire:click="test">wire:click</button>
</div>
```

The first portion (using $wire) doesn't work, the second portion does

**Comment by SimoTod** [comment]
It's worth noting that support for Alpine v3 in livewire was added last
night so you might have to update

**Reply by hendrasob** [reply]
Yes that's right, you better update support for Alpine v3.

**Reply by calebporzio** [reply]
Yep, v2.5.1 is the release

**Reply by nkeena** [reply]
I've upgraded to Livewire 2.5.1 and and pulled in the Alpine v3.1 using the CDN and I still can't get `@entangle` to work.

```
// blade view
<div x-data="{
    tab: @entangle('tab')
}">
```

```
// livewire class
public $tab = 'general';
```

**Reply by SimoTod** [reply]
Livewire needs to be loaded before Alpine (or Alpine needs to be deferred). Is that your issue?
